### PR TITLE
chore(project) upgrade deps; remove Babel runtime requirement

### PR DESCRIPTION
**Pull request by bhj**
Hey, was able to get this most of the way there, and the built lib works (fixing #3) but the tests timeout for some reason. Mind taking a look? Thanks!

### CODE REVIEW

**Comment by bhj** [comment]
@jedwards1211 any thoughts?

File: .travis.yml [hunk 1]
```diff
@@ -5,13 +5,12 @@ cache:
 notifications:
   email: false
 node_js:
-  - '8'
-  - '7'
-  - '6'
+  - '14'
+  - '12'
 install:
-  - yarn install --ignore-engines --check-files --frozen-lockfile
```
Comment (jedwards1211) [inline .travis.yml:12]: yarn works in travis, is there some reason you changed this?

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "flow:coverage": "for file in src/**.js test/**.js; do echo $file; flow coverage $file; done",
     "flow:watch": "flow-watch -e js,js.flow,flowconfig --ignore lib/ --ignore node_modules/ --watch .flowconfig --watch src/ --watch test/",
     "build": "rimraf lib && babel src --out-dir lib && flow-copy-source -v src/ lib",
-    "test": "NODE_ENV=production BABEL_ENV=test nyc --reporter=lcov --reporter=text mocha $npm_package_config_mocha",
+    "test": "NODE_ENV=production BABEL_ENV=test nyc --require @babel/register --reporter=lcov --reporter=text mocha $npm_package_config_mocha",
```
Comment (jedwards1211) [inline package.json:14]: I've never needed to `--require @babel/register` for `nyc` like this, requiring it in `mocha` has always been sufficient

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
       "src/**/*.js"
     ],
     "require": [
-      "babel-register"
+      "@babel/register"
```
Comment (jedwards1211) [inline package.json:36]: oh I forgot about this in the nyc config.  Huh.  In any case if this is necessary the `--require @babel/register` on the command line should be redundant

**Comment by jedwards1211** [comment]
I think the test failures have to do with `clock.tick()` from `sinon.useFakeTimers` not being able to cause promises to resolve

**Comment by jedwards1211** [comment]
btw I just discovered, Sinon now has `await clock.tickAsync()`

**Pull request closed without merging**